lib/collection/variable-changeset.js

```diff
@@ -0,0 +1,235 @@
+ *   "value": "Bob"
+ * }
+ */
+
+/**
+ * A serialized version of JSON changeset. This format is extended from the JSON delta specification.
+ *
+ * This format allows capturing variable creates, updates and deletes in a very minimal and consistent format.
+ * It has 3 parts.
+ *  ID
+ *  Key path
+ *  Value
+ *
+ * This format diverges from the original JSON delta format by adding an id segment. This is to allow mapping meta data
+ * to changes later.
+ *
+ * For example ['1', ['name'], 'Bob'] represents an update on the variable `name` to `Bob`. This can be extended for
+ * other types of changes as well. So ['1', ['description'], 'Hi there!'] could mean an addition if `description`
+ * is not present already.
+ *
+ * Unset, or delete of a variable can be captured by not specifying the `value` segment. e.g. ['1', ['age']] would mean
+ * a delete for the variable `age`.
+ *
+ *
+ * See {@link http://json-delta.readthedocs.io/en/latest/philosophy.html} for more info.
+ *
+ * @typedef {Array} VariableChangeset~serialized
+ *
+ * @property {Number|String} 0
+ * @property {String[]} 1
+ * @property {*} 2
+ */
+_.inherit((
+
+    /**
+     * An instruction representing a change to a variable.
+     *
+     * @constructor
+     * @param {VariableChangeset~definition|VariableChangeset~serialized} [definition]
+     */
+    VariableChangeset = function VariableChangeset (definition) {
+        definition = definition || {};
+
+        // definition is a serialized changeset
+        if (_.isArray(definition)) {
+            definition = deserializeChangeset(definition);
+        }
+
+        // this constructor is intended to inherit and as such the super constructor is required to be executed
+        VariableChangeset.super_.call(this, definition);
+
+        // import definition
+        this.update(definition);
+    }
+), Property);
+
+_.assign(VariableChangeset.prototype, /** @lends VariableChangeset.prototype */ {
+    /**
+     * Defines that this property requires an ID field
+     * @private
+     * @readOnly
+     */
+    _postman_propertyRequiresId: true,
+
+    /**
+     * Updates the variable changeset.
+     *
+     * @param {VariableChangeset~definition} definition
+     */
+    update: function (definition) {
+        _.mergeDefined(this, /** @lends VariableChangeset.prototype */ {
+            /**
+             * @property {String[]}
+             */
+            key: _.isArray(definition.key) ? definition.key.join('.') : definition.key || '',
+
+            /**
+             * @property {String}
+             */
+            type: definition.type || INSTRUCTION.UNSET,
+
+            /**
+             * @property {*}
+             */
+            value: definition.type === INSTRUCTION.SET ? definition.value : undefined
+        });
+    },
+
+    /**
+     * Applies the changeset on a variable scope
+     *
+     * @param {VariableScope} scope
+     */
+    applyOn: function (scope) {
+        if (!(scope && isVariableScope(scope))) {
```

Comment:
no point checking for isVariableScope. Diff expects an object with set / unset / etc (duck typed. even duck checking would not be needed IMHO)

Reply:
It makes sense to not check for `VariableScope`.